api/test/ProgramsInteract.test.js

```diff
@@ -0,0 +1,160 @@
+const { readFileSync, readdirSync } = require('fs');
+const { join } = require('path');
+const yaml = require('js-yaml');
+const { GearApi, GearKeyring, getWasmMetadata } = require('../lib');
+
+const EXAMPLES_DIR = 'test/wasm';
+const programs = new Map();
+const testFiles = readdirSync('test/spec/programs');
+const api = new GearApi();
+const accounts = {
+  alice: GearKeyring.fromSuri('//Alice'),
+  bob: GearKeyring.fromSuri('//Bob'),
+};
+const testif = (condition) => (condition ? test : test.skip);
+
+jest.setTimeout(20000);
+
+beforeAll(async () => {
+  await api.isReady;
+});
+
+afterAll(async () => {
+  await api.disconnect();
+  await new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, 2000);
+  });
+});
+
+for (let filePath of testFiles) {
+  /**
+   * @type {{title: string, programs: {name: string, id: number, gasLimit: number, value: number, account: string, meta: boolean, initPayload: any}[], messages: {id: number, program: number, payload: any, gasLimit: number, value: number, log: string}[], gasSpent: number[]}}
+   */
+  const testFile = yaml.load(readFileSync(join('./test/spec/programs', filePath), 'utf8'));
+
+  describe(testFile.title, () => {
+    testif(!testFile.skip)('Upload programs', async () => {
+      for (let program of testFile.programs) {
+        const code = readFileSync(join(EXAMPLES_DIR, `${program.name}.opt.wasm`));
+        const metaFile = readFileSync(join(EXAMPLES_DIR, `${program.name}.meta.wasm`));
+        const meta = program.meta ? await getWasmMetadata(metaFile) : {};
+        const programId = api.program.submit(
+          { code, initPayload: program.initPayload, gasLimit: program.gasLimit, value: program.value },
+          meta,
+        );
+        programs.set(program.id, {
+          id: programId,
+          meta,
+          metaFile,
+        });
+        let unsub, log, messageId;
+
+        if (program.log) {
+          log = new Promise(async (resolve) => {
```

Comment:
async callback in promise initialisation maybe could be rewritten?

Reply:
https://github.com/gear-tech/gear-js/pull/268/commits/1e4ef14b6c297dfff315f3503574abe3a7293eac

That's better?

Reply:
Yep, thank you.